index.js

```diff
@@ -47,8 +47,32 @@ const isObject = value => value !== null && typeof value === 'object';
 const supportsAbortController = typeof globals.AbortController === 'function';
 const supportsStreams = typeof globals.ReadableStream === 'function';
 
+const mergeHeaders = (...sources) => {
+	const result = {};
+
+	for (const source of sources) {
+		if (!isObject(source)) {
+			throw new TypeError('The `headers` argument must be an object');
+		}
+
+		const headers = new globals.Headers(source);
+
+		for (const [key, value] of headers) {
+			// Headers constructor changes the value to a string
+			if (value === 'undefined' || typeof value === 'undefined') {
```

Comment:
Replace these conditions with `value === undefined`

Reply:
It should be mentioned in the readme that `{something: 'undefined'}` and `{something: undefined}` will be omitted.

Reply:
IMO, `'undefined'` should only be omitted if it comes from a `Headers` instance.

Reply:
> IMO, 'undefined' (as a string) should only be omitted if it comes from a Headers instance.

Sure, as long as it is mentioned in the readme :)